spack-environments/blt-test/toss3-clang10.yaml

```diff
@@ -0,0 +1,154 @@
+###############################################################################
+# Copyright (c) 2019-2021, Lawrence Livermore National Security, LLC and other
+# RADIUSS-CI project contributors. See top-level COPYRIGHT file for details.
+#
+# SPDX-License-Identifier: MIT
+###############################################################################
+
+spack:
+  view: false
+  concretization: separately
+  specs:
+  - blttest
```

Comment:
One thing I've been wondering is how we can support multiple specs within an environment, e.g. when we add more compilers.

Would we do something like:
```yaml
spack:
  concretization: separately
  specs:
  - blttest%clang@10.0.0
  - blttest%gcc@8.3.1
 ```

Or would we have one environment per compiler/spec ?

Reply:
> One thing I've been wondering is how we can support multiple specs within an environment, e.g. when we add more compilers.
> 
> Would we do something like:
> 
> ```yaml
> spack:
>   concretization: separately
>   specs:
>   - blttest%clang@10.0.0
>   - blttest%gcc@8.3.1
> ```
> 
> Or would we have one environment per compiler/spec ?

You could use Spack stacks (https://spack-tutorial.readthedocs.io/en/latest/tutorial_stacks.html), where you'd define specs and define compilers and Spack would build each spec with each compiler.

Reply:
You can see a relatively simple example at https://github.com/LLNL/radiuss-spack-testing/blob/dahlgren1/add-care-env/spack-environments/care/spack.yaml.

Reply:
I think there is value in testing individual spack envs - since that is most likely how users will build when developing. But again, that is tied to my deployment model where I use a config to dial everything in to deploy with a specific set of features on a specific platform, and then I use multiple configs when I need more than one deployments (for example, cuda only, vs openmp only). Maybe thats not an important goal for the blt testing.

Reply:
You can also specify multiple `matrix`s.   There's a somewhat simple example at https://github.com/mpbelhorn/olcf-spack-environments/blob/develop/hosts/lyra/envs/base/spack.yaml.